src/sources/http_scrape/integration_tests.rs

```diff
@@ -0,0 +1,229 @@
+//! Integration tests for http_scrape source.
+//! The container configuration file is `docker-compose-.http_scrape.yml`
+//! It leverages a static file server which serves the files in tests/data/http-scrape
+
+use tokio::time::{Duration, Instant};
+
+use crate::{
+    config::{ComponentKey, SourceConfig, SourceContext},
+    http::Auth,
+    serde::default_framing_message_based,
+    sources::http_scrape::scrape::NAME,
+    tls,
+    tls::TlsConfig,
+    SourceSender,
+};
+use codecs::decoding::DeserializerConfig;
+use vector_core::config::log_schema;
+
+use super::{
+    tests::{run_compliance, run_error, INTERVAL_SECS},
+    HttpScrapeConfig,
+};
+
+/// Logs (raw bytes) should be scraped and decoded successfully.
+#[tokio::test]
+async fn scraped_logs_bytes() {
+    let events = run_compliance(HttpScrapeConfig::new(
+        "http://dufs:5000/logs/bytes".to_string(),
+        INTERVAL_SECS,
+        None,
+        DeserializerConfig::Bytes,
+        default_framing_message_based(),
+        None,
+        None,
+        None,
+    ))
+    .await;
+    let log = events[0].as_log();
+    assert_eq!(log[log_schema().source_type_key()], NAME.into());
+}
+
+/// Logs (json) should be scraped and decoded successfully.
+#[tokio::test]
+async fn scraped_logs_json() {
+    let events = run_compliance(HttpScrapeConfig::new(
+        "http://dufs:5000/logs/json.json".to_string(),
+        INTERVAL_SECS,
+        None,
+        DeserializerConfig::Json,
+        default_framing_message_based(),
+        None,
+        None,
+        None,
+    ))
+    .await;
+    let log = events[0].as_log();
+    assert_eq!(log[log_schema().source_type_key()], NAME.into());
+}
+
+/// Metrics should be scraped and decoded successfully.
+#[tokio::test]
+async fn scraped_metrics_native_json() {
+    let events = run_compliance(HttpScrapeConfig::new(
+        "http://dufs:5000/metrics/native.json".to_string(),
+        INTERVAL_SECS,
+        None,
+        DeserializerConfig::NativeJson,
+        default_framing_message_based(),
+        None,
+        None,
+        None,
+    ))
+    .await;
+
+    let metric = events[0].as_metric();
```

Comment:
I was going to say we should assert that the `Event` type is `Metric` here but I _think_ `as_metric()` would bubble that up?

Reply:
Correct, from `lib/vector-core/src/event/ref.rs` : 

```Rust
      /// Extract the `Metric` reference in this.
      ///
      /// # Panics
      ///
      /// This will panic if this is not a `Metric` reference.
      pub fn as_metric(self) -> &'a Metric {
          match self {
              Self::Metric(metric) => metric, &mut Metric
              _ => panic!("Failed type coercion, {:?} is not a metric reference", self),
          }
      }
```

Reply:
Thanks for double checking!

Reply:
👍 added a comment